roles/gluster_hci/tasks/hosted_engine_pre_check.yml

```diff
@@ -0,0 +1,55 @@
+---
+# Check if 1MB write with 512byte chunk size takes less than 10 seconds.
+# The writes are to be done on GlusterFS mounts
+
+- set_fact:
+      mount_points: "{{ mount_points|default([]) +
+                        [ '/mnt/tmp/' + item.volname ] }}"
+  with_items: "{{ gluster_features_hci_volumes }}"
+
+- set_fact:
+      glusterfs_server: "{{ gluster_features_hci_cluster[0] }}"
+
+- name: Check the latency of the disk before HE install
+  block:
+     - name: Mount the glusterfs volume
+       mount:
+          path: "/mnt/tmp/{{ item.volname }}"
+          state: mounted
+          fstype: glusterfs
+          src: "{{ glusterfs_server }}:{{ item.volname }}"
+       run_once: true
+       with_items: "{{ gluster_features_hci_volumes }}"
+
+     # We do a awk print NF-3 since newer version of coreutils print data copied both
+     # in MB and MiB thus adding extra column, which will break our time extraction
+     # command.
+     - name: Check the disk latency (mount points)
+       shell: >
+          dd if=/dev/zero of={{item}}/small.file bs=512 count=2050 2>&1 |
```

Comment:
That's wrong - it uses the cache, since there's no oflag=direct.

Also note that dd is not very high performance, and writing zeros, when VDO is on the backend, is going to be very different than regular IO.
But it's a reasonable start. We can enhance it later with fio (as was done in the Colonizer code)

Reply:
Yeah. I missed oflag=direct, that's my mistake. I'll add oflag=direct for now.